src/lib.rs

```diff
@@ -166,11 +182,17 @@ impl DesktopEntry {
         // If appid matches
         id == self.id()
             // or startup_wm_class matches
-            || self.startup_wm_class().map_or(false, |wm_class| wm_class == id)
+            || self.startup_wm_class().is_some_and(|wm_class| wm_class == id)
+            // or X-Ubuntu-Gettext-Domain was set
+            || self.ubuntu_gettext_domain.as_ref().is_some_and(|domain| *domain == id)
             // or the path itself matches
             || self.path.file_stem()
                 .and_then(|os_str| os_str.to_str())
-                .map_or(false, |name| name == id)
+                .is_some_and(|name| {
+                    name == id
+                        // Or match by last part of app ID
+                        || id.split('.').rev().next().is_some_and(|id| id == name)
+                })
```

Comment:
not in spec as well (see first comment).

Reply:
Unsure how else to match `cool-retro-term.cool-retro-term`

Reply:
Popsicle also appears as `Popsicle`, but there's no exact match for it other than `com.system76.Popsicle`.

Reply:
> Popsicle also appears as Popsicle, but there's no exact match for it other than com.system76.Popsicle.

Well that is on us then, we should fix that. Still wondering how this works elsewhere.